papis/commands/add.py

```diff
@@ -455,6 +456,10 @@ def run(paths: List[str],
             "existing document.")
 
         # NOTE: we always want the user to confirm if a duplicate is found!
+        # Except when batch option is set
+        if batch:
+            logger.warning("No document has been created, since batch option exit when a duplicate is found.")
+            return
         confirm = True
```

Comment:
```suggestion
        # except when batch option is set
        confirm = not batch
```
? Is this what you intended here? The `return` in your current code definitely shouldn't be there.

Reply:
Yes! Better. I am not used to python I did'nt think about this trick. The goal was simple : when run with the batch option, add should never overwrite an existing entry.

Reply:
This won't do that, as far as I know. `confirm = True` means asking the user if the new document should be added and doesn't consider duplicates. In effect, this will always *overwrite* when `batch = True`.

Reply:
You think? It is strange because it works when I test it. But that's true than the 'confim' options has not really this purpose.
But we agree than this block of code is about the duplication checking?? Maybe we can just return when the batch option is set?

```diff
    try:
        found_document = papis.utils.locate_document_in_lib(tmp_document)
    except IndexError:
...
    else:
+        if batch:
+            return
        click.echo("The following document is already in your library:")
...
        # NOTE: we always want the user to confirm if a duplicate is found!
        confirm = True
```

By the way, I just remember than in fact batch option set confirm always to false:
```
609-613:
    if batch:
        edit = False
        confirm = False
        open_file = False
```

Reply:
Yeah, you're completely right! 

However, I'm not sure that never overwriting is a good choice. Overwriting duplicates with the new version seems like a reasonable way to go as well, so we might want to add a flag for that?

Reply:
well, maybe, in fact, now, it is not overwriting the entry but adding another one (a duplicate), its a problem, after you have twice the same item in you library, its a mess. I don't think many people want that. But we can ad a flag yes if you think its a good thing... 
I think overwriting the entry with add is useless, as the info message tells you:
"Hint: Use the 'papis update' command instead to update the existing document."
To 'overwrite' an existing entry with new data, we can use `update`.
To add another entrey (duplicate of another one), we can use `add`. 
Now, is there a case a guy wants to add a duplicate with `add -b` ? Quite rare I think, who 1) wants twice the same entry 2) has to run `add` in the background?
Remember than we have to make a choice. We can let `add -b` prompt the user to ask confirmation, as we lose the all point of --batch option : running thing quietly in the background. 
If really you think its necessary, lets add a flag --force, forcing adding the entry in any case, and then:

```python
        if batch and not force: 
            logger.info("No document is created.") 
            return
```

Reply:
> well, maybe, but in fact, now, it is not overwriting the entry but adding another one (a duplicate), its a problem, after you have twice the same item in you library, its a mess. 

Hm, if it actually does that, that's a bug indeed. It shouldn't add the same document twice.

> To 'overwrite' an existing entry with new data, we can use `update`. To add another entry (duplicate of another one), we can use `add`. 

That's a good point, we do have the `update` command to do that, so maybe the `add` command doesn't have to be that smart. I'll have to think about it a bit :confused:

Reply:
Yeah, I've come around on this too. Overwriting is a bit mean and we don't know how to do a smart merge of the new and existing documents if they're "duplicates", so better just let the user know and skip that document.

Thanks for your patience thinking through this! :heart: